controllers/servers/csi/volume_group_server.py

```diff
@@ -63,15 +63,15 @@ def DeleteVolumeGroup(self, request, _):
             return volumegroup_pb2.DeleteVolumeGroupResponse()
 
         array_type = volume_group_id_info.array_type
-        volume_group_id = volume_group_id_info.ids.internal_id
+        volume_group_name = volume_group_id_info.ids.name
         array_connection_info = utils.get_array_connection_info_from_secrets(secrets)
 
         with get_agent(array_connection_info, array_type).get_mediator() as array_mediator:
             logger.debug(array_mediator)
 
             try:
-                logger.debug("Deleting volume group {}".format(volume_group_id))
-                array_mediator.delete_volume_group(volume_group_id)
+                logger.debug("Deleting volume group {}".format(volume_group_name))
+                array_mediator.delete_volume_group(volume_group_name)
```

Comment:
deleting by id is not safer?

Reply:
id is a running number - like "3", it is not unique enough, this way we can be sure we are deleting the right volume. if we had cross storage unique id for volume-group that could be the best.
if we delete by id we can accidentally delete the next volume group that catches the id (on retry, for example)